### PR TITLE
Add collidedWithMap and add GameData to collided calls

### PR DESCRIPTION
**Pull request by jakob753951**

### CODE REVIEW

**Review by MGertz — APPROVED** [review]
LGTM

**Review by The0mikkel — CHANGES_REQUESTED** [review]
Missing some JavaDoc

File: Common/src/main/java/dk/sdu/sesem4/common/data/entity/Entity.java [hunk 1]
```diff
@@ -121,6 +121,10 @@ public void setEntityType(EntityType entityType) {
           this.entityType = entityType;
      }
 
+	public void collided(GameData gameData, Entity other) {
```
[the remark this answers is not in the document]
Comment (jakob753951) [reply Common/src/main/java/dk/sdu/sesem4/common/data/entity/Entity.java:130]: JavaDoc has been added :)

File: Common/src/main/java/dk/sdu/sesem4/common/data/entity/Entity.java [hunk 2]
```diff
@@ -142,6 +146,11 @@ public void collided(Entity other) {
 		doDamage(other);
 		doKnockback(other);
 	}
+	
+	public void collidedWithMap(GameData gameData) {
```
[the remark this answers is not in the document]
Comment (jakob753951) [reply Common/src/main/java/dk/sdu/sesem4/common/data/entity/Entity.java:163]: JavaDoc has been added :)

**Review by Muhammed-als — APPROVED** [review]
LGTM

**Review by The0mikkel — APPROVED** [review]

**Comment by The0mikkel** [comment]
Please wait for #119 (Sprint 3) to be merged, before merging this one

**Review by JfriisJ — APPROVED** [review]
looks good

**Review by MGertz — APPROVED** [review]

**Review by The0mikkel — APPROVED** [review]

**Pull request merged**